src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -559,6 +583,14 @@ public int getMaxTotalUses() {
         return maxTotalUses;
     }
 
+    public List<NodePropertyDescriptor> getNodePropertyDescriptors() {
+        return NodePropertyDescriptor.for_(NodeProperty.all(), EC2AbstractSlave.class);
+    }
+
+    public DescribableList<NodeProperty<?>, NodePropertyDescriptor> getNodeProperties() {
+        return Objects.requireNonNull(nodeProperties);
```

Comment:
Nit: it might make more sense to throw an `IllegalStateException` here instead of `NullPointerException`.

Reply:
Done! WDYT?